module/zfs/dmu_send.c

```diff
@@ -3019,38 +3042,13 @@ receive_object_range(struct receive_writer_arg *rwa,
 	if (drror->drr_firstobj > rwa->max_object)
 		rwa->max_object = drror->drr_firstobj;
 
-	offset = drror->drr_firstobj * sizeof (dnode_phys_t);
-	mdn = DMU_META_DNODE(rwa->os);
```

Comment:
For clarity please add a comment here briefly describing why this needs to be deferred until `receive_object()`.  Something like this (stolen from your commit message) would be good.

```
/*
 * The DRR_OBJECT_RANGE handling must be deferred to receive_object()
 * so that the encryption parameters are set with each object that is
 * written into that block.
 */
```